src/main/java/org/mule/service/http/impl/service/util/RedirectUtils.java

```diff
@@ -51,35 +66,47 @@ public static boolean isRedirected(int statusCode) {
   }
 
   /**
-   * @param requestMethod
-   * @param responseStatusCode
-   * @return the original request method if the status code is 301, 307 or 308, or GET if it is 302 or 303.
+   * Copy from RedirectHandler#sendAsGet.
+   * 
+   * @param statusCode
+   * @return false if the status code is 301, 307 or 308, or true if it is 302 or 303.
    */
-  public static String getMethodForStatusCode(String requestMethod, int responseStatusCode) {
-    switch (responseStatusCode) {
-      case 301:
-      case 307:
-      case 308:
-        return requestMethod;
-      case 302:
-      case 303:
-        return HttpConstants.Method.GET.name();
-      default:
-        throw new IllegalArgumentException("Invalid status code");
-    }
+  private boolean mustSendAsGet(int statusCode) {
+    return !(statusCode < 302 || statusCode > 303) && !(statusCode == 302 && isStrict302Handling);
```

Comment:
why not just test for 303?

Reply:
If the status code is 302 or 303 the redirect request must be sent as GET, except if the status code is 302 and the 302 strict handlings is activated. I copy this code from grizzly: 
https://github.com/eclipse-ee4j/grizzly-ahc/blob/master/src/main/java/com/ning/http/client/providers/grizzly/AhcEventFilter.java#L795